onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -372,23 +321,38 @@ bool BaseOpBuilder::HasSupportedInputs(
 }
 
 bool BaseOpBuilder::IsOpSupportedImpl(ModelBuilder& /* model_builder */,
-                                      const ONNX_NAMESPACE::NodeProto& /* node */) {
+                                      const onnxruntime::Node& /* node */) {
   return true;
 }
 
 void BaseOpBuilder::AddToModelBuilder(ModelBuilder& model_builder,
-                                      const ONNX_NAMESPACE::NodeProto& node) {
+                                      const onnxruntime::Node& node) {
   ORT_ENFORCE(IsOpSupported(model_builder, node),
-              "Unsupported operator " + node.op_type());
+              "Unsupported operator " + node.OpType());
 
   AddToModelBuilderImpl(model_builder, node);
-  LOGS_DEFAULT(VERBOSE) << "Operator name: [" << node.name()
-                        << "] type: [" << node.op_type() << "] was added";
+  LOGS_DEFAULT(VERBOSE) << "Operator name: [" << node.Name()
+                        << "] type: [" << node.OpType() << "] was added";
 }
 
-void BaseOpBuilder::AddToModelBuilderImpl(ModelBuilder& /* model_builder */,
-                                          const ONNX_NAMESPACE::NodeProto& node) {
-  ORT_NOT_IMPLEMENTED("Unsupported operator " + node.op_type());
+bool BaseOpBuilder::HasExternalInitializer(ModelBuilder& model_builder,
+                                           const onnxruntime::Node& node) {
+  const auto& initializers(model_builder.GetOnnxGraph().GetAllInitializedTensors());
+  for (const auto* node_arg : node.InputDefs()) {
```

Comment:
Would probably be quicker to create a set of external intializer names once and just checking against that.

Reply:
I guess this will be true if we have some reused initializers, will keep this as is for now since I do not really want to add another member variable to the ModelBuilder


---
In reply to: [448725831](https://github.com/microsoft/onnxruntime/pull/4389#discussion_r448725831) [](ancestors = 448725831)